src/Aspire.Dashboard/Extensions/ResourceViewModelExtensions.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Dashboard.Model;
+
+namespace Aspire.Dashboard.Extensions;
+
+public static class ResourceViewModelExtensions
+{
+    internal static KnownResourceState? GetKnownState(this ResourceViewModel resource)
```

Comment:
Can we put this on the view model instead of parsing it all the time?

Reply:
done